subsys/bluetooth/audio/codec.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2022 Bose Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/** @file
+ *  @brief Bluetooth LE-Audio codec LTV parsing
+ *
+ *  Helper functions to parse codec config data as specified in the Bluetooth assigned numbers for
+ *  Generic Audio.
+ */
+
+#include <bluetooth/audio/audio.h>
+#include <sys/byteorder.h>
+
+/* Overkill for now, but useful if data format is changed from bt_codec_data to TLV format. */
+void bt_codec_parse(const struct bt_codec *codec,
+		   bool (*func)(const struct bt_codec_data *data, void *user_data),
+		   void *user_data)
+{
+	for (int i = 0; i < codec->data_count; i++) {
+		if (!func(&codec->data[i], user_data)) {
+			return;
+		}
+	}
+}
+
+void bt_meta_parse(const struct bt_codec *codec,
+		   bool (*func)(const struct bt_codec_data *data, void *user_data),
+		   void *user_data)
+{
+	for (int i = 0; i < codec->meta_count; i++) {
+		if (!func(&codec->meta[i], user_data)) {
+			return;
+		}
+	}
+}
+
+bool bt_codec_get_val(const struct bt_codec *codec,
```

Comment:
```suggestion
static bool bt_codec_get_val(const struct bt_codec *codec,
```

Reply:
This should be public, added to API - thanks for highlighting!